tensorflow/lite/kernels/fully_connected.cc

```diff
@@ -64,6 +64,49 @@ constexpr int kBiasTensor = 2;
 constexpr int kOutputTensor = 0;
 constexpr int kShuffledInputWorkspaceTensor = 1;
 
+inline TfLiteStatus CheckTypes(TfLiteContext* context,
+                               const TfLiteTensor* input,
+                               const TfLiteTensor* filter,
+                               const TfLiteTensor* bias, TfLiteTensor* output,
+                               TfLiteFullyConnectedParams* params) {
+  const bool is_Quantized =
+      ((filter->type == kTfLiteUInt8) || (filter->type == kTfLiteInt8));
+  const bool is_Hybrid = is_Quantized && (input->type == kTfLiteFloat32);
+  const bool is_Shuffled =
+      is_Quantized && (params->weights_format ==
+                       kTfLiteFullyConnectedWeightsFormatShuffled4x16Int8);
+  const bool is_BiasNotFloatType = bias && (bias->type != kTfLiteFloat32);
+  const bool is_BiasNotIntType = bias && (bias->type != kTfLiteInt32);
```

Comment:
I find these flags and checking that they are false somewhat hard to reason about, because of the double negation.

Can we instead do

// optional bias tensor.
const bool is_optional_int_bias = !bias || (bias->type == kTfLiteInt32);
const bool is_optional_float_bias = !bias || (bias->type == kTfLiteFloat32);

Then the checks below will be ensure true.